src/Data/Managers/V2/GetInstanceWithRelations/useGetInstanceWithRelations.ts

```diff
@@ -0,0 +1,91 @@
+import { UseQueryResult, useQuery } from "@tanstack/react-query";
+import { ServiceInstanceModel } from "@/Core";
+import { PrimaryBaseUrlManager } from "@/UI";
+import { useFetchHelpers } from "../helpers";
+
+/*
+ * interface for the service instance with its related instances and eventual coordinates on canvas
+ */
+export interface InstanceWithReferences {
+  instance: ServiceInstanceModel;
+  relatedInstances: InstanceWithReferences[];
+  coordinates?: string;
+}
+
+interface GetInstanceWithRelationsHook {
+  useOneTime: () => UseQueryResult<InstanceWithReferences, Error>;
+}
+
+/**
+ * React Query hook to fetch an instance with its related instances from the API.
+ * @param {string} id - The ID of the instance to fetch.
+ * @param {string} environment - The environment in which we are looking for instances.
+ * @returns  {GetInstanceWithRelationsHook} An object containing a custom hook to fetch the instance with its related instances.
+ */
+export const useGetInstanceWithRelations = (
+  id: string,
```

Comment:
instanceId, I don't think we use simply id in other places for this 🤔

Reply:
we use instanceId below, but I changed places then